app/controllers/provider_interface/sessions_controller.rb

```diff
@@ -3,14 +3,58 @@ class SessionsController < ProviderInterfaceController
     skip_before_action :authenticate_provider_user!
     skip_before_action :check_data_sharing_agreements
 
-    def new; end
+    def new
+      if FeatureFlag.active?('dfe_sign_in_fallback')
+        render :authentication_fallback
+      end
+    end
 
     def destroy
       DfESignInUser.end_session!(session)
 
       redirect_to action: :new
     end
 
+    def sign_in_by_email
+      raise unless FeatureFlag.active?('dfe_sign_in_fallback')
+
+      provider_user = ProviderUser.find_by(email_address: params.dig(:provider_user, :email_address))
+
+      if provider_user
+        magic_link_token = MagicLinkToken.new
+        ProviderMailer.fallback_sign_in_email(provider_user, magic_link_token.raw).deliver_later
+        provider_user.update!(magic_link_token: magic_link_token.encrypted, magic_link_token_sent_at: Time.zone.now)
+
+        SlackNotificationWorker.perform_async(
+          "Provider user #{provider_user.first_name} has requested a fallback signin link",
+          edit_support_interface_provider_user_path(provider_user),
+        )
+      end
+    end
+
+    def authenticate_with_token
+      magic_link_token = MagicLinkToken.from_raw(params.fetch(:token))
+      provider_user = ProviderUser.find_by!(magic_link_token: magic_link_token)
```

Comment:
Presumably we would check for token expiry at this stage. It doesn't look like we've built that yet which seems OK

Reply:
Added to https://trello.com/c/t421tPrM/2073-deal-with-may-1st-dfe-signin-outage.